internal/service/provisioning/api/openapi.yaml

```diff
@@ -261,26 +319,42 @@ components:
         templateParameters:
           type: object
           description: Input Parameters that conform to the OpenAPI V3 schema defined in the template.
+          x-go-type: k8sruntime.RawExtension
+          x-go-type-import:
+            path: k8s.io/apimachinery/pkg/runtime
+            name: k8sruntime
         status:
           type: object
           description: Status of the provisioning request.
+          x-go-type-name: ProvisioningRequestStatus
           properties:
             provisioningStatus:
               type: object
               description: Details about the status of the provisioning request.
+              x-go-type-name: ProvisioningStatus
               properties:
                 updateTime:
                   type: string
-                  format: date-time
                   description: |
                     Timestamp indicating the last time the status of the provisioning request was updated.
+                  x-go-type: metav1.Time
+                  x-go-type-import:
+                    path: k8s.io/apimachinery/pkg/apis/meta/v1
+                    name: metav1
                   example: 2042-07-21T20:32:28Z
+                  x-go-type-skip-optional-pointer: true
                 message:
                   type: string
                   description: Message describing the status of the provisioning request.
+                  x-go-type-skip-optional-pointer: true
                 provisioningState:
                   type: string
                   description: Current state of the provisioning request.
+                  x-go-type: provisioningv1alpha1.ProvisioningState
```

Comment:
I'm a little worried about the use of these custom go types in the API specification.  This may cause some pain to clients that are code generating their client code based on our swagger yaml.  I would prefer if we avoided these custom types and used the standard swagger ones as much as possible and dealt with any conversions ourselves in our backend implementation.

cc:  @pixelsoccupied

Reply:
that's true `x-` vendor extension starts to reduce portability. But generally it would be noop since "officially" openapi says if a generator cant recognize vendor extension it should just ignore it? But `x-go..` reduces portability further since now using Go is sort of required by the clients (unless there's something codegen also handles...maybe treats as `object` type?...e.g what happens if want to generate python client?) 

But using`x-go-type` makes life so much easier! I personally be ok but yeah i leave it up to you.

Reply:
@alegacy @pixelsoccupied Thank you for pointing that out. You bring up a very reasonable concern. Yes, my original thinking using x-go-type was to avoid conversions in our backend. However, I now see how this could introduce complications for clients relying on standard tools for code generation, especially in other languages. I will change to use the standard types.